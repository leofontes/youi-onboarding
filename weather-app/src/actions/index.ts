import * as tmdbActions from './tmdbActions';
import * as youtubeActions from './youtubeActions';

export {
  tmdbActions as tmdb,
  youtubeActions as youtube,
};
