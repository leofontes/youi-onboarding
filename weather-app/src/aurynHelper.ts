import { DeviceInfo, Ref } from '@youi/react-native-youi';
import { NativeModules, findNodeHandle } from 'react-native';
import { once } from 'lodash';

const systemName = DeviceInfo.getSystemName();
const { Cloud, RefUtils } = NativeModules;

interface AurynHelper {
  hasHardwareBackButton: boolean;
  isRoku: boolean;
  updateCloudScene: (component: React.RefObject<any>) => void;
  togglePointerEvents: (ref: Ref, enabled: boolean) => boolean;
  toggleParentPointerEvents: (ref: Ref, enabled: boolean) => boolean;
}

export const AurynHelper: AurynHelper = {
  hasHardwareBackButton: !['iOS'].includes(systemName),
  isRoku: Cloud.isCloudServer,
  updateCloudScene: (component) => {
    if (!Cloud.isCloudServer || !component.current) return;
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), false);
      Cloud.sendFocusMap();
    }, 300);
  },
  togglePointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
  toggleParentPointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setParentCompositionPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
};
