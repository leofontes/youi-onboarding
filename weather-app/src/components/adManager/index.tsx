import React, { Component, createRef, RefObject } from 'react';
import { View, StyleSheet } from 'react-native';

import PauseAd from './pause-ad';
import { AdContextType, AdContext, AdState } from './context';
import LowerThirdAd from './lower-third-ad';

interface AdManagerProps {
  pauseAdCompositionName?: string;
  pauseAdClosed: () => void;
}

class AdManager extends Component<AdManagerProps> {
  declare context: AdContextType;

  static contextType = AdContext;

  static defaultProps = {
    pauseAdClosed: () => {},
  };

  private pausAdRef: RefObject<PauseAd> = createRef();

  handleOnClose = () => this.context.setPauseAdState(AdState.closed);

  renderPauseAd = () => {
    const { pauseAdCompositionName } = this.props;

    if (pauseAdCompositionName === undefined) return null;

    this.context.setPauseAdEnabled(true);

    return <PauseAd ref={this.pausAdRef} name={pauseAdCompositionName} onClose={this.handleOnClose} />;
  };

  renderLowerThirdAd = () => {
    return <LowerThirdAd />;
  }

  render() {
    return <View style={styles.container}>
      {this.renderPauseAd()}
      {this.props.children}
      {this.renderLowerThirdAd()}
    </View>;
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    ...StyleSheet.absoluteFillObject,
  },
});

export default AdManager;
