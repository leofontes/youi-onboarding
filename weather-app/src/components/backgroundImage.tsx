import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

type BackgroundImageProps = {
  source: string;
};

export const BackgroundImage: React.FunctionComponent<BackgroundImageProps> = ({source, children}) =>
  <View style={styles.container}>
    <Image style={styles.background} source={{ uri: source }} />
    {children}
  </View>

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#0D0D0D',
  },
  background: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    bottom: 0,
  },
});
