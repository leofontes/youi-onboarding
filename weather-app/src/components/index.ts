export { BackButton } from './backButton';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { NavigationBar } from './navigationBar';
export { TvContainer } from './tvContainer';
export { Error } from './error';
export { withOrientation } from './withOrientation';
export { LiveContainer } from './liveContainer';
export { LiveListItem } from './liveListitem';
export { BackgroundImage } from './backgroundImage';
