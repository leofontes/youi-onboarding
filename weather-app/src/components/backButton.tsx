import React from 'react';
import { ButtonRef, ButtonRefProps } from '@youi/react-native-youi';
import { AurynHelper } from '../aurynHelper';

export const BackButton: React.FunctionComponent<Partial<ButtonRefProps>> = ({ focusable, ...otherProps }) => (
  <ButtonRef
    {...otherProps}
    name="Btn-Back"
    visible={!AurynHelper.hasHardwareBackButton}
    focusable={focusable && !AurynHelper.hasHardwareBackButton}
  />
);
