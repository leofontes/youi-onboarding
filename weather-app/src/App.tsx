import React from 'react';
import store from './store';
import Stack from './navigation';
import { FormFactor } from '@youi/react-native-youi';
import { Provider } from 'react-redux';
import { BackgroundImage } from './components';

export default class YiReactApp extends React.Component {
  render = () =>
    <Provider store={store}>
      <BackgroundImage source={!FormFactor.isHandset ? 'res://drawable/default/CES-auryn-10ft-bg.png' : 'res://drawable/default/Gradient-Bottom-2.png'}>
        <Stack />
      </BackgroundImage>
    </Provider>
}
