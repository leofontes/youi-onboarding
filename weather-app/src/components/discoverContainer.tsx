/* eslint-disable max-len */

import React from 'react';
import { View } from 'react-native';
import { ListItem } from '.';
import { Asset } from '../adapters/asset';
import { ListItemPressEvent, ListItemFocusEvent } from './listitem';
import { FormFactor } from '@youi/react-native-youi';
import { AdListItem } from './adManager/list-item-ad';

interface DiscoverContainerProps {
  data: Asset[];
  onPressItem?: ListItemPressEvent;
  onFocusItem?: ListItemFocusEvent;
  focusable?: boolean;
  index: number;
}

// eslint-disable-next-line max-lines-per-function
export const DiscoverContainer: React.FunctionComponent<DiscoverContainerProps> = ({
  data,
  onPressItem,
  onFocusItem,
  focusable,
  index,
}) => {
  if (data.length !== 3) return null;

  const smallItems = (
    <View style={{ flexDirection: 'row', justifyContent: FormFactor.isHandset ? 'space-between' : 'flex-start' }}>
      <ListItem
        focusable={focusable}
        onPress={onPressItem}
        onFocus={onFocusItem}
        shouldChangeFocus={index % 2 !== 0}
        imageType={{ type: 'Backdrop', size: 'Small' }}
        data={data[0]}
      />
      {index !== 1 ? (
        <ListItem
          focusable={focusable}
          onPress={onPressItem}
          onFocus={onFocusItem}
          shouldChangeFocus={index % 2 !== 0}
          imageType={{ type: 'Backdrop', size: 'Small' }}
          data={data[1]}
        />
      ) : (
        <AdListItem focusable={focusable} onPress={onPressItem} onFocus={onFocusItem} />
      )}
    </View>
  );
  const largeItem = (
    <ListItem
      focusable={focusable}
      onPress={onPressItem}
      onFocus={onFocusItem}
      shouldChangeFocus={index % 2 === 0}
      imageType={{ type: 'Backdrop', size: 'Large' }}
      data={data[2]}
    />
  );
  if (index % 2) {
    return (
      <View style={{ backgroundColor: 'transparent' }}>
        {smallItems}
        {largeItem}
      </View>
    );
  }

  return (
    <View style={{ backgroundColor: 'transparent' }}>
      {largeItem}
      {smallItems}
    </View>
  );
};
