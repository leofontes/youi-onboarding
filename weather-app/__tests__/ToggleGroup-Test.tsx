
import React from 'react';
import { ToggleGroup, ToggleButton } from '../src/components';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<ToggleGroup
    onPressItem={jest.fn()}
    initialToggleIndex={0}>
    <ToggleButton/>
    <ToggleButton/>
    <ToggleButton/>
  </ToggleGroup>).toJSON();
  expect(tree).toMatchSnapshot();
});
