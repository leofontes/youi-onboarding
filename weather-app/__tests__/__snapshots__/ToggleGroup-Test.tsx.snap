// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <YiPushButtonView
    name="Btn-Nav-List"
    onCompositionDidLoad={[Function]}
    onFocus={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      autoplay={true}
      direction="forward"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
  </YiPushButtonView>,
  <YiPushButtonView
    name="Btn-Nav-List"
    onCompositionDidLoad={[Function]}
    onFocus={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      autoplay={false}
      direction="reverse"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
  </YiPushButtonView>,
  <YiPushButtonView
    name="Btn-Nav-List"
    onCompositionDidLoad={[Function]}
    onFocus={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      autoplay={false}
      direction="reverse"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
  </YiPushButtonView>,
]
`;
